Bottom: 7634e8cb0b91c13bd622548ee6555f53bfb6de33
Top:    bc5df1dc5fd2230e206819c453b126891aafa6f1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-24 02:25:38 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index d06030d..bcb9a52 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
